Src/FluentAssertions/Common/Guard.cs

```diff
@@ -2,12 +2,13 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Runtime.CompilerServices;
+using JetBrains.Annotations;
 
 namespace FluentAssertions.Common;
 
 internal static class Guard
 {
-    public static void ThrowIfArgumentIsNull<T>([ValidatedNotNull] T obj,
+    public static void ThrowIfArgumentIsNull<T>([ValidatedNotNull][NoEnumeration] T obj,
```

Comment:
🤔 I'm starting to wonder I like all these attributes. What do you think?

Reply:
Let's see how this evolves. It if becomes too noisy, we can always decide to disable the rule and remove the attributes.